addons/mail/models/ir_binary.py

```diff
@@ -0,0 +1,32 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models
+from odoo.exceptions import AccessError
+
+
+class IrBinary(models.AbstractModel):
+    _inherit = "ir.binary"
+
+    def _find_record_check_access(self, record, access_token, field, **kwargs):
+        try:
+            return super()._find_record_check_access(record, access_token, field, **kwargs)
+        except AccessError:
+            if (
+                record._name in ["res.partner", "mail.guest"]
+                and field == "avatar_128"
+                and kwargs.get("thread_model")
+                and kwargs.get("thread_id")
+            ):
+                thread = self.env[kwargs.pop("thread_model")]._get_thread_with_access(
+                    int(kwargs.pop("thread_id")), **kwargs
+                )
```

Comment:
We can discuss about it to find a compromise that works for everyone, the method should probably be moved to models instead of being on ir binary, this way each model can add its own behavior without the risk of impacting other more sensitive models such as attachments in particular.